uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -11,6 +11,71 @@ define([
   instances.factory('LeaveRequestInstance', ['$q', 'checkPermissions', 'OptionGroup',
     'shared-settings', 'ModelInstance', 'LeaveRequestAPI',
     function ($q, checkPermissions, OptionGroup, sharedSettings, ModelInstance, LeaveRequestAPI) {
+      /**
+       * Amends balance change breakdown if calculation unit absence type is "hours".
+       * It changes the balance first and last days (only first day in case of single day request)
+       *   by setting the selected deductions.
+       * @NOTE this function mutates the "balanceChange" object
+       *
+       * @param  {Object} balanceChange
+       * @param  {String} calculationUnit
+       * @return {Promise} resolve with amended balance change in case of "hours" absence type
+       *   or returns unamended balance change in case of "days" absence type
+       */
+      function amendHourlyBalanceChangeBreakdown (balanceChange, calculationUnit) {
+        var breakdown = balanceChange.breakdown;
+        var request = this;
+
+        if (calculationUnit !== 'hours') {
+          return balanceChange;
+        }
+
+        return OptionGroup.valuesOf('hrleaveandabsences_leave_request_day_type')
+          .then(function (dayTypes) {
+            if (balanceChange) {
```

Comment:
If there is no balance change, you are not doing anything, so
it should be
```
        if (calculationUnit !== 'hours' || !balanceChange) {
          return balanceChange;
        }

        return OptionGroup.valuesOf('hrleaveandabsences_leave_request_day_type')
          .then(function (dayTypes) {
            amendHourlyBalanceChangeForDay.call(request, _.first(_.values(breakdown)), 'from', dayTypes);

            if (breakdown.length > 1) {
              amendHourlyBalanceChangeForDay.call(request, _.last(_.values(breakdown)), 'to', dayTypes);
            }

            amendHourlyBalanceChangeAmount(balanceChange);
          });
```

Reply:
if there is no balance change the function should not even be called

Reply:
Balance change is an object returned from API, it always exists.

Reply:
If that is the case ` if (balanceChange) {`check is not necessary?

Reply:
Done.